parsl/executors/high_throughput/executor.py

```diff
@@ -659,8 +662,20 @@ def submit(self, func, resource_specification, *args, **kwargs):
         Returns:
               Future
         """
-
-        validate_resource_spec(resource_specification, self.enable_mpi_mode)
+        if self.enable_mpi_mode:
+            validate_resource_spec(resource_specification, self.enable_mpi_mode)
+        else:
+            if resource_specification and isinstance(resource_specification, dict):
```

Comment:
this all belongs in validate_resource_spec()

try implementing https://github.com/Parsl/parsl/issues/3519 as a separate PR first to get the structure in place in that function for "mpi vs non-mpi" resource validation (with, I guess, no resource spec allowed for non-mpi mode?) and you can get a parsl issue fixed as a nice side effect